src/Screens/Main/Chat/index.tsx

```diff
@@ -17,18 +18,35 @@ import { Dialog, DialogProps } from '../../components/Dialog';
 
 import colors from '../../components/colors';
 import { BanAction } from 'src/api/buddies';
+import { MentorCardExpandedRoute } from '../MentorCardExpanded';
 
 export type ChatRoute = {
   'Main/Chat': { buddyId: string };
 };
 
-type Props = navigationProps.NavigationProps<ChatRoute, ChatRoute>;
+type Props = navigationProps.NavigationProps<
+  ChatRoute,
+  ChatRoute & MentorCardExpandedRoute
+>;
 
 const Chat = ({ navigation }: Props) => {
   const goBack = () => {
     navigation.goBack();
   };
 
+  const buddyId = navigation.getParam('buddyId');
+
+  const mentor = ReactRedux.useSelector(mentorState.getMentorByUserId(buddyId));
+
+  const goToMentorCard = () => {
+    if (mentor) {
+      navigation.navigate('Main/MentorCardExpanded', {
+        mentor,
+        fromChat: true,
```

Comment:
"didNavigateFromChat" would make it more certain that is is boolean. From chat could be Id or something. this is just fine as it is tho.